### PR TITLE
Remove maven actions to follow Apache policies

### PR DESCRIPTION
**Pull request by ricardozanini**
In this PR:

- Remove setup maven and maven settings XML actions to include custom bash scripts to do the same thing
- Review actions to cache maven download and fetch this repo for a settings.xml reference file and a script to install maven

TODO:
- [x] - Test on my fork

### CODE REVIEW

**Comment by jstastny-cz** [comment]
We're adapting for Apache projects, is there expectation to support also non transfered kiegroup projects? If so, we should somehow parameterize repository information. 
If it's only for Apache projects, then we could replace the URLs.

Otherwise looks good, installation of maven could be done using sdkman :-) (if you have any reservations to its use in CI I'd like to learn.)

I don't have any insight into before/after comparison of the configuration done using these actions, but seems legit.

**Comment by ricardozanini** [comment]
It's working now, you can see it here: https://github.com/ricardozanini/java-helloworld/actions

@jstastny-cz thanks! Replies inline.

> We're adapting for Apache projects, is there expectation to support also non transfered kiegroup projects? If so, we should somehow parameterize repository information.

Not sure what you mean. I don't think we need to parameterize repository information, can you give more information? Where, exactly?

> installation of maven could be done using sdkman

I don't have any restrictions, we can do it using whatever, including the GHA Java Action, but since we need a specific Maven version ('cause related/dependency projects), Tristan used a third-party action. If we can specify a version and have it added to classpath, feel free to add a commit to this PR using sdkman.

I just would like to move forward fast to avoid being blocked by CI. :/

**Comment by jstastny-cz** [comment]
All good, for the repositories I meant in the Maven settings template - I saw there jboss nexus URLs. This should be changed to Apache nexus URLs or parameterized by the actions config?

**Comment by ricardozanini** [comment]
> All good, for the repositories I meant in the Maven settings template - I saw there jboss nexus URLs. This should be changed to Apache nexus URLs or parameterized by the actions config?

OH SURE

We can update the template in a later PR adding Apache URLs! :D

**Comment by jstastny-cz** [comment]
I would suggest replacing as part of this one to prevent possible use of stale snapshots. There is no need to have jboss URL there already.

**Comment by ricardozanini** [comment]
> I would suggest replacing as part of this one to prevent possible use of stale snapshots. There is no need to have jboss URL there already.

Cool! Can you give me the updated URLs or just add a suggestion to change here? I understand that this is a super simple change to do, right? Thank u!

File: .ci/actions/maven/action.yml [hunk 1]
```diff
@@ -48,70 +65,11 @@ runs:
...
-              }
-            }
-          ]
-        plugin_groups: >
-          [
-            "org.zanata"
-          ]
+      shell: bash
+      run: |
+        cp maven-tools/.ci/actions/maven/settings-template.xml ${HOME}/.m2/settings.xml
```
Comment (baldimir) [inline .ci/actions/maven/action.yml:70]: I think the custom settings.xml generation can be removed completely. Each repository has the configuration of the repositories in their root pom, e.g. here it is in kogito-runtimes https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/pom.xml#L75. I think that way the deployment etc. can be controlled more easily, than in a custom action like this. What do you think please?

**Review by baldimir — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]
LGTM. Let's follow the PR executions once this is in place and closely investigate any differences in PR checks outcome GHA vs Jenkins.

**Pull request merged**